Amazon.IonDotnet/Internals/Text/SystemTextReader.cs

```diff
@@ -576,15 +576,15 @@ public override IEnumerable<SymbolToken> GetTypeAnnotationSymbols()
 
         public override bool HasAnnotation(string annotation)
         {
-            if (annotation == null)
-            {
-                throw new ArgumentNullException(nameof(annotation));
-            }
-
             int? annotationId = null;
             foreach (SymbolToken symbolToken in _annotations)
             {
-                if (symbolToken.Text == null)
+                //zero symbol scenario
+                if (annotation == null && symbolToken.Text == null && symbolToken.Sid == 0)
+                {
+                    return true;
+                }
+                else if (symbolToken.Text == null)
```

Comment:
Nit: the symbolToken.Text == null check exists twice here, so this can be reordered to be a bit better.

Reply:
Addressed